Event.cpp

```diff
@@ -195,39 +197,39 @@ constexpr uint32_t RECORD_HEADER_SIZE(uint16_t num_fields, uint16_t name_size, u
 
 constexpr uint32_t FIELD_NAME_SIZE_OFFSET = 0;
 constexpr uint32_t FIELD_NAME_SIZE_SIZE = sizeof(uint16_t);
-inline uint16_t& FIELD_NAME_SIZE(uint8_t* data, uint32_t field_offset) {
-    return *reinterpret_cast<uint16_t*>(data+field_offset);
+inline uint16_t& FIELD_NAME_SIZE(uint8_t* data, uint32_t record_offset, uint32_t field_offset) {
```

Comment:
In general, do we need to do bound check when we casting to avoid overflow.

Reply:
Any overflow would be due to programmer error, not invalid/malicious input.